src/cluster_legacy.c

```diff
@@ -6591,10 +6618,11 @@ int clusterAsmOnEvent(const char *task_id, int event, void *arg) {
                     clusterAddSlot(myself, j);
                 }
             }
-            /* New config and Bump new config */
+            /* Bump config epoch and broadcast the new config to the other nodes. */
             clusterBumpConfigEpochWithoutConsensus();
-            clusterSaveConfigOrDie(1);
-            clusterBroadcastPong(CLUSTER_BROADCAST_ALL);
+            clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG|
```

Comment:
Before we actually call two `fsync` twice,  `clusterBumpConfigEpochWithoutConsensus` also set the flag `clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG|CLUSTER_TODO_FSYNC_CONFIG);`